src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala

```diff
@@ -108,8 +108,8 @@ class HtsjdkRecordReader(val callFields: Set[String]) extends Serializable {
         if (hasGT) {
           val i = vc.getAlleleIndex(a0)
           val j = vc.getAlleleIndex(a1)
-          val gt = Genotype.gtIndexWithSwap(i, j)
-          rvb.addInt(gt)
+          val gt = Genotype.diploidGtIndexWithSwap(i, j)
+          rvb.addInt(Call2.fromUnphasedDiploidIndex(gt))
```

Comment:
Do you need to create gt here?  Can't you just construct the correct call?  I feel like we might be able to get rid of `diploidGtIndexWithSwap` and just call the relevant call constructors.